hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java

```diff
@@ -0,0 +1,179 @@
+import org.hibernate.mapping.Collection;
+import org.hibernate.persister.collection.BasicCollectionPersister;
+import org.hibernate.persister.spi.PersisterCreationContext;
+import org.hibernate.pretty.MessageHelper;
+import org.hibernate.reactive.pool.ReactiveConnection;
+import org.hibernate.reactive.session.ReactiveConnectionSupplier;
+import org.hibernate.reactive.util.impl.CompletionStages;
+
+import org.jboss.logging.Logger;
+
+public class ReactiveBasicCollectionPersister extends BasicCollectionPersister implements ReactiveCollectionPersister {
+	private static final CoreMessageLogger LOG = Logger.getMessageLogger(
+			CoreMessageLogger.class,
+			ReactiveBasicCollectionPersister.class.getName()
+	);
+	private static final String VALUE_STR = "(?, ?)";
+
+	public ReactiveBasicCollectionPersister(
+			Collection collectionBinding,
+			CollectionDataAccess cacheAccessStrategy,
+			PersisterCreationContext creationContext) throws MappingException, CacheException {
+		super( collectionBinding, cacheAccessStrategy, creationContext );
+	}
+
+	private ReactiveConnection getReactiveConnection(SharedSessionContractImplementor session) {
+		return ( (ReactiveConnectionSupplier) session ).getReactiveConnection();
+	}
+
+	public CompletionStage<Void> recreateReactive(
+			PersistentCollection collection,
+			Serializable id,
+			SharedSessionContractImplementor session)
+			throws HibernateException {
+		ReactiveConnection reactiveConnection = getReactiveConnection( session );
+
+		if ( isInverse ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( !isRowInsertEnabled() ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( LOG.isDebugEnabled() ) {
+			LOG.debugf(
+					"Inserting collection: %s",
+					MessageHelper.collectionInfoString( this, collection, id, session )
+			);
+		}
+
+		// create all the new entries
+		Iterator entries = collection.entries( this );
+
+		if ( entries.hasNext() ) {
+			collection.preInsert( this );
+
+			List<Object> valuePairList = new ArrayList<>();
+
+			while ( entries.hasNext() ) {
+				valuePairList.add( id );
+				valuePairList.add( entries.next() );
+			}
+
+			// In order to insert element collection values create a native query with placeholders followed by
+			// value sets
+			return reactiveConnection
+					.update(
+							getSQLInsertElementCollectionString(
+									valuePairList.size() / 2 ),
+							valuePairList.toArray( new Object[0] )
+					)
+					.thenAccept( ignore -> {
+					} );
+		}
+
+		return CompletionStages.voidFuture();
+	}
+
+	// FIXME: this method currently only works for List<String> element collections and uses a VALUE_STR = "(?, ?)"
+	// will not work for List, Map, etc... of @Embedded entity (i.e. complex type)
+	private String getSQLInsertElementCollectionString(int numRows) {
+		String defaultInsertString = getSQLInsertRowString();
+		StringBuilder buf = new StringBuilder( defaultInsertString.length() + ( numRows + 2 ) * VALUE_STR.length() );
```

Comment:
Are you replacing VALUE_STR because the vert.x driver has a different syntax for parameters?

Reply:
If that's the case I don't think this is the right approach, we should replace the parameters when the query is created in `getSQLInsertRowString()`.

Reply:
Do you even need this part? I think we already do the conversion of the parameters in: [SqlConnection#prepartedQuery](https://github.com/hibernate/hibernate-reactive/blob/master/hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/SqlClientConnection.java#L163)

Reply:
Right, parameter placeholder replacement is handled by the connection layer. We don't need to do it here.

Reply:
k.. thx for the direction

Reply:
> If that's the case I don't think this is the right approach, we should replace the parameters when the query is created in `getSQLInsertRowString()`.

getSQLInsertRowString() just returns a generated/cached AbstractCollectionPersister string created in it's constructor.

Reply:
Note these strings are what you see when you enable SQL logging (i.e. you see placeholders  .. (?, ?..) etc.)

Reply:
Well.. yes, it's printing the query with the placeholders for the parameters.
Not sure why in this specific case you are trying to do something different